
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Install and Load these packages
```{r}

if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install()
BiocManager::install(c("BiocParallel", "DelayedArray","vsn","GenomeInfoDb","GenomicRanges","DESeq2","SummarizedExperiment","DEGreport","rain","geneplotter","biclust","DESeqDataFromMatrix"))

install.packages(c("DEGreport","patchwork","data.table","ggplot2","lattice","colorspace","grid","MASS","stats","graphics","grDevices","utils","datasets","methods","base","readr"))

library(c("BiocParallel", "DelayedArray","vsn","GenomeInfoDb","GenomicRanges","DESeq2","SummarizedExperiment","DEGreport","rain","geneplotter","biclust","DESeqDataFromMatrix","DEGreport","patchwork","data.table","ggplot2","lattice","colorspace","grid","MASS","stats","graphics","grDevices","utils","datasets","methods","base","readr"))

```

Import Data Sets
```{r}
#Ta8 Count Data
ta8_raw_counts_formatted <- read_delim("~/ta8_raw_counts_formatted.tsv", 
    "\t", escape_double = FALSE, trim_ws = TRUE)
#Mo1 Count Data
raw_counts_formated <- read_delim("~/raw_counts_formated.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)

#Ta8 Column Data
ta8_col_data <- read_delim("~/ta8_rc_formated_col_conditions.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)
#Mo1 Colum Data
col_data <- read_delim("~/raw_counts_formated_col_conditions.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)
```

Format the Raw data 
```{r}
###FORMAT Ta8 RAWCOUNT TABLE###
ta8_rawcount<-as.matrix(ta8_raw_counts_formatted)
names(ta8_rawcount[ , sort(names(ta8_rawcount))])
row.names(ta8_rawcount)<-ta8_raw_counts_formatted$locus_tag 
ta8_rawcount <- ta8_rawcount[,2:29]
ta8_rawcount<-sapply(ta8_rawcount, FUN=as.integer)
ta8_rawcount<-matrix(data=ta8_rawcount, ncol = 28, nrow=1364)
row.names(ta8_rawcount)<-ta8_raw_counts_formatted$locus_tag
ta8_coldata_matrix <- as.matrix(ta8_rc_formated_col_conditions)
ta8_col_data<-ta8_coldata_matrix[,2:3]
row.names(ta8_col_data) <- ta8_coldata_matrix[,1]
colnames(ta8_rawcount)<-colnames(ta8_raw_counts_formatted[,2:29])
ta8_rawcount<-ta8_rawcount[,order(match(colnames(ta8_rawcount),rownames(ta8_col_data)))]

###FORMAT Mo1 RAWCOUNT TABLE###
mo1_rawcount<-as.matrix(raw_counts_formated)
row.names(mo1_rawcount)<-raw_counts_formated$locus_tag
mo1_rawcount <- mo1_rawcount[,2:25]
mo1_rawcount<-sapply(mo1_rawcount, FUN=as.integer)
mo1_rawcount<-matrix(data=mo1_rawcount, ncol = 24, nrow=1746)
row.names(mo1_rawcount)<-raw_counts_formated$locus_tag
coldata_matrix <- as.matrix(raw_counts_formated_col_conditions)
col_data<-coldata_matrix[,2:3]
row.names(col_data) <- coldata_matrix[,1]
colnames(mo1_rawcount)<-rownames(col_data)


```

Format/Edit the metadata table
```{r}
###Add metadata to ta8_col_data###
ta8_col_data<-cbind(ta8_col_data, c("0","0","0","0","15","15","15","15","60","60","60","60","360","360","360","360","735","735","735","735","780","780","780","780","1080","1080","1080","1080"))
ta8_col_data<-cbind(ta8_col_data, c("dark","dark","dark","dark","light","light","light","light","light","light","light","light","light","light","light","light","dark","dark","dark","dark","dark","dark","dark","dark","dark","dark","dark","dark"))
ta8_col_data<-cbind(ta8_col_data, c("0","0","0","0","15","15","15","15","60","60","60","60","360","360","360","360","15","15","15","15","60","60","60","60","360","360","360","360"))
ta8_col_data<-cbind(ta8_col_data, c("0","0","0","0","1","1","1","1","2","2","2","2","3","3","3","3","4","4","4","4","5","5","5","5","6","6","6","6"))
colnames(ta8_col_data)<-c("condition", "grouping", "time", "ld_status","time_since_change","timepoint")


###Add metadata to Mo1 col_data###
col_data<-cbind(col_data, c("0","0","0","15","15","15","60","60","60","360","360","360","360","735","735","735","735","780","780","780","1080","1080","1080","1080"))
col_data<-cbind(col_data, c("dark","dark","dark","light","light","light","light","light","light","light","light","light","light","dark","dark","dark","dark","dark","dark","dark","dark","dark","dark","dark"))
col_data<-cbind(col_data, c("dark:0","dark:0","dark:0", "light:15","light:15","light:15", "light:60","light:60","light:60", "light:360","light:360","light:360","light:360", "dark:735","dark:735","dark:735","dark:735", "dark:780","dark:780","dark:780", "dark:1080","dark:1080","dark:1080","dark:1080"))
colnames(col_data)<-c("condition", "replicate", "time", "ld_status", "group")

```

PCA Analysis and Plots
```{r}
###TA8 PCA Analysis###
ta8_rawcount_edit<-subset(ta8_rawcount, select=c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28))
ta8_col_data_edit<-ta8_col_data[c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28),]
ta8_dds_edit<-DESeqDataSetFromMatrix(countData = ta8_rawcount_edit, colData = ta8_col_data_edit, design = ~condition)
ta8_dds_edit<-DESeq(ta8_dds_edit)
ta8_dds_edit_res<-results(ta8_dds_edit)
ta8_dds_edit_rlog<-rlog(ta8_dds_edit_res)
ta8_dds_edit_rlog_false<-rlog(ta8_dds_edit, blind=FALSE)
ta8_dds_edit_vst<-vst(ta8_dds_edit, blind=FALSE)

plotpcadatata8<-plotPCA(ta8_dds_edit_rlog, intgroup=c("ld_status","time"), returnData=TRUE) 
percentVarta8 <- round(100 * attr(plotpcadatata8, "percentVar"))
plotpcadatata8$group <- factor(plotpcadatata8$group, levels = c("dark:0", "light:15", "light:60", "light:360", "dark:735", "dark:780", "dark:1080"))

pca_patchwork1<-ggplot(plotpcadatata8, aes(x=PC1, y=PC2, color=group, shape=group))+
geom_point(size=6)+
scale_color_manual( name = "Sample", values=c("black","#F3BFB2","#E77E65","#C3492C", "#ADAEE4","#5B5CC8", "#20219F"), labels=c("dark, 0 hours, t0", "light, 0.25 hours, t1", "light, 1 hour, t2", "light, 6 hours, t3", "dark, 12.25 hours, t4", "dark, 13 hours, t5", "dark, 18 hours, t6"))+
scale_shape_manual( name = "Sample", values=c(1,17,17,17,19,19,19), labels=c("dark, 0 hours, t0", "light, 0.25 hours, t1", "light, 1 hour, t2", "light, 6 hours, t3", "dark, 12.25 hours, t4", "dark, 13 hours, t5", "dark, 18 hours, t6")) +
# scale_alpha_manual(values=c("F"=0, "M"=1)) +
xlab(paste("PC1: ", percentVarta8[1], "% variance")) +
ylab(paste("PC2: ", percentVarta8[2], "% variance")) +
ggtitle("A.")+
theme_bw()+ 
theme(text = element_text(size = 16),legend.position = "none")

###Mo1 PCA Analysis###
mo1_dds<- DESeqDataSetFromMatrix(countData = mo1_rawcount, colData = col_data, design = ~condition)
mo1_dds <-DESeq(mo1_dds)
mo1_res_unedited<-results(mo1_dds)
mo1_dds_rlog<-rlog(mo1_dds)

###TAKE OUT SAMPLES DUE TO PCA##
mo1_rawcount_edit<-subset(mo1_rawcount, select=c(1,2,3,4,5,6,7,8,9,10,12,13,14,16,17,18,20,21,23,24))
mo1_col_data_edit<-col_data[c(1,2,3,4,5,6,7,8,9,10,12,13,14,16,17,18,20,21,23,24),]
mo1_dds_edit<- DESeqDataSetFromMatrix(countData = mo1_rawcount_edit, colData = mo1_col_data_edit, design = ~condition)

mo1_dds_edit<-DESeq(mo1_dds_edit)
mo1_res<-results(mo1_dds_edit)
mo1_dds_edit_rlog<-rlog(mo1_dds_edit)

###MO1 PCA PLOTTING###

### UNEDITED PCA: Sample Not Removed###
plotpcadatamo1<-plotPCA(mo1_dds_rlog, intgroup=c("ld_status","time"), returnData=TRUE)
# 
###EDITED PCA: Sample Removed##
 plotpcadatamo1_edit<-plotPCA(mo1_dds_edit_rlog, intgroup=c("ld_status","time"), returnData=TRUE) 

percentVarmo1 <- round(100 * attr(plotpcadatamo1, "percentVar"))
plotpcadatamo1$group <- factor(plotpcadatamo1$group, levels = c("dark:0", "light:15", "light:60", "light:360", "dark:735", "dark:780", "dark:1080"))

pca_patchwork2<-ggplot(plotpcadatamo1, aes(x=PC1, y=PC2, color=group, shape=group))+
geom_point(size=6)+
scale_color_manual( name = "Sample", values=c("black","#F3BFB2","#E77E65","#C3492C", "#ADAEE4","#5B5CC8", "#20219F"), labels=c(expression("dark, 0 hours, t"[0]), expression("light, 0.25 hours, t"[1]), expression("light, 1 hour, t"[2]), expression("light, 6 hours, t"[3]), expression("dark, 12.25 hours, t"[4]),expression("dark, 13 hours, t"[5]),expression("dark, 18 hours, t"[6])))+
scale_shape_manual( name = "Sample", values=c(1,17,17,17,19,19,19), labels=c(expression("dark, 0 hours, t"[0]), expression("light, 0.25 hours, t"[1]), expression("light, 1 hour, t"[2]), expression("light, 6 hours, t"[3]), expression("dark, 12.25 hours, t"[4]),expression("dark, 13 hours, t"[5]),expression("dark, 18 hours, t"[6]))) +
# scale_alpha_manual(values=c("F"=0, "M"=1)) +
xlab(paste("PC1: ", percentVarmo1[1], "% variance")) +
ylab(paste("PC2: ", percentVarmo1[2], "% variance")) +
#ggtitle("B.")+
theme_bw()+ 
theme(text = element_text(size = 16),
      #legend.position = c(.845,.5),
      legend.position = "right",
      legend.background = element_blank(),
      legend.box.background = element_rect(colour = "black"))

pca_patchwork3<-ggplot(plotpcadatamo1_edit, aes(x=PC1, y=PC2, color=group, shape=group))+
geom_point(size=6)+
scale_color_manual( name = "Sample", values=c("black","#F3BFB2","#E77E65","#C3492C", "#ADAEE4","#5B5CC8", "#20219F"), labels=c(expression("dark, 0 hours, t"[0]), expression("light, 0.25 hours, t"[1]), expression("light, 1 hour, t"[2]), expression("light, 6 hours, t"[3]), expression("dark, 12.25 hours, t"[4]),expression("dark, 13 hours, t"[5]),expression("dark, 18 hours, t"[6])))+
scale_shape_manual( name = "Sample", values=c(1,17,17,17,19,19,19), labels=c(expression("dark, 0 hours, t"[0]), expression("light, 0.25 hours, t"[1]), expression("light, 1 hour, t"[2]), expression("light, 6 hours, t"[3]), expression("dark, 12.25 hours, t"[4]),expression("dark, 13 hours, t"[5]),expression("dark, 18 hours, t"[6]))) +
# scale_alpha_manual(values=c("F"=0, "M"=1)) +
xlab(paste("PC1: ", percentVarmo1[1], "% variance")) +
ylab(paste("PC2: ", percentVarmo1[2], "% variance")) +
#ggtitle("B.")+
theme_bw()+ 
theme(text = element_text(size = 16),
      #legend.position = c(.845,.5),
      legend.position = "right",
      legend.background = element_blank(),
      legend.box.background = element_rect(colour = "black"))

###GENERATE PCA PLOTS###

(pca_patchwork1 + pca_patchwork3)

ggsave("PCA_plot_a_and_c.tiff", width=85, device='tiff', units = "mm",dpi=300, scale=4)

###UNEDITED MO1 PCA###
pca_patchwork2

ggsave("PCA_plot_MO1_pre_sample_removal.tiff", width=85, device='tiff', units = "mm",dpi=300, scale=4)
  


```
Run DESEQ,
Rlog tranform data,
Run LRT,
Run Hierarchical Clustering, 
Make Cluster Plots
```{r}
###Run DESEQ###
ta8_dds<- DESeqDataSetFromMatrix(countData = ta8_rawcount, colData = ta8_col_data, design = ~condition)
ta8_dds <-DESeq(ta8_dds)
ta8_res<-results(ta8_dds)

###Log2 Transform Ta8 Count Data###
rld_ta8_dds<-rlog(ta8_dds, blind=TRUE)
###MAKE A SummarizedExperiment object of rlog transformed data###
rld_mat_ta8_dds<-assay(rld_ta8_dds)

###Run LRT Test###
ta8_dds_update<-DESeq(ta8_dds, test="LRT", reduced = ~1)
res_update<-results(ta8_dds_update)

sig_res_LRT_ta8<-res_update %>%
                  data.frame() %>%
                  rownames_to_column(var="gene")%>%
                  as_tibble() %>%
                  filter(padj < 0.015)
clustering_sig_res_LRT_ta8<-sig_res_LRT_ta8 %>%
  arrange(padj)

###Write P-values of LRT to file###
write.csv(as.data.frame(cbind(rownames(res_update),res_update$padj)), "ta8_LRT_padj_values.csv")

###Hierarchically Cluster Significant Differentially Expressed Genes###
cluster_rlog_sig_res_LRT_ta8<-rld_mat_ta8_dds[clustering_sig_res_LRT_ta8$gene, ]
clusters_sig_res_LRT_ta8<-degPatterns(cluster_rlog_sig_res_LRT_ta8, metadata = meta, col = "timepoint", plot = T, minc = 1)

###Calculate Z-score of Ta8DE Genes###
cluster_rlog_sig_res_LRT_ta8_zscore<-t(scale(t(cluster_rlog_sig_res_LRT_ta8), scale=TRUE, center=TRUE))

LRT_ta8_sampling_timepoints<-c(-5,-5,-5,-5,15,15,15,15,60,60,60,60,360,360,360,360,735,735,735,735,780,780,780,780,1080,1080,1080,1080)

clusters_sig_res_LRT_ta8_vector<-clusters_sig_res_LRT_ta8$df$cluster
names(clusters_sig_res_LRT_ta8_vector)<-clusters_sig_res_LRT_ta8$df$genes

###Format Data to Plot in ggplot###
LRT_ta8_data <- data.table(melt(data.table(class = as.factor(clusters_sig_res_LRT_ta8_vector), cluster_rlog_sig_res_LRT_ta8_zscore)))
LRT_ta8_data[, Time := rep(LRT_ta8_sampling_timepoints, each = nrow(cluster_rlog_sig_res_LRT_ta8_zscore)) ]
LRT_ta8_data[, ID := rep(1:nrow(cluster_rlog_sig_res_LRT_ta8_zscore), ncol(cluster_rlog_sig_res_LRT_ta8_zscore))]

LRT_ta8_data$facet = factor(LRT_ta8_data$class, levels = c("9", "7", "8","5", "10","13", "1", "4","2", "3", "12", "6", "11", "14"))
LRT_ta8_data$Time<-as.factor(LRT_ta8_data$Time)

panel.LRT_labs_ta8<- c("A.", "B.","C.", "D.", "E.", "F.","G.", "H.","I.", "J.", "K.", "L.", "M.", "N.")
names(panel.LRT_labs_ta8)<-c("9", "7", "8","5", "10","13", "1", "4","2", "3", "12", "6", "11", "14")

LRT_panel_labeller_ta8 <- function(variable,value){
  return(panel.LRT_labs_ta8[value])
}

###Summarize Clusters and Format Summary for Plotting###
lrt_summary<-summarySE(LRT_ta8_data, measurevar="value", groupvars=c("class","Time"))
lrt_summary$Time=as.numeric(levels(lrt_summary$Time))[lrt_summary$Time]
lrt_summary$facet = factor(lrt_summary$class, levels = c("9", "7", "8","5", "10","13", "1", "4","2", "3", "12", "6", "11", "14"))
lrt_summary$annotation = c("N = 127", rep("",6),"N = 64",rep("",6),"N = 55", rep("",6),"N = 73", rep("",6),"N = 91", rep("",6),"N = 9",rep("",6),"N = 91", rep("",6),"N = 42",rep("",6),"N = 81", rep("",6),"N = 34", rep("",6),"N = 26", rep("",6),"N = 12", rep("",6),"N = 7", rep("",6),"N = 17",rep("",6))
lrt_summary$gene_group = c("G", rep("",6),"I",rep("",6),"J", rep("",6),"H", rep("",6),"D", rep("",6),"L",rep("",6),"B", rep("",6),"C",rep("",6),"A", rep("",6),"E", rep("",6),"M", rep("",6),"K", rep("",6),"F", rep("",6),"N",rep("",6))


###LINE PLOT OF HIERARCHICAL CLUSTERS###
lrt_summary_plot_ta8<-ggplot(lrt_summary, aes(x=Time, y=value)) +
  facet_wrap(~facet, ncol = 6, scales = "fixed", drop=FALSE) +
  geom_rect(data=NULL,aes(xmin=-Inf,xmax=0,ymin=-Inf,ymax=Inf), fill="gray")+
  geom_rect(data=NULL,aes(xmin=0,xmax=720,ymin=-Inf,ymax=Inf), fill="white")+
  geom_rect(data=NULL,aes(xmin=720,xmax=Inf,ymin=-Inf,ymax=Inf), fill="gray")+
  geom_ribbon(aes(ymin=value-sd, ymax=value+sd), alpha=.2) +
  geom_line() +
  geom_point()+
  labs(x = "Time (Hours)", y = "Z-score of transcript abundance") +
  scale_x_continuous(breaks=c(0,360,720,1080), labels = c("0","6","12","18"), position = "top") +
  theme(
  strip.text = element_blank(),
  text = element_text(size=25),
  legend.position = "none",
  strip.background =element_rect(fill="white"),
  panel.background = element_rect(fill="white"))+
  geom_text(aes(x=900, y=-2, label=annotation), size=6)+
  geom_text(aes(x=70, y=2.5, label=gene_group), size=8)
    
ggsave("ta8_LRT_clusters_line_plot_sd_ribbon_pvalue_0.015.tiff", width=85, height = 45, device='tiff', units = "mm",dpi=300, scale=7)

##########################
###Mo1 LRT Clusters PLOT##
##########################

###Add metadata to the edited mo1_col_data###
mo1_meta<-cbind(mo1_col_data_edit, c("1","1","1","2","2","2","3","3","3","4","4","4","5","5","5","6","6","7","7","7"))
colnames(mo1_meta)<-c("condition","replicate","time","ld_status","group","timepoint")

###Run DESEQ###
mo1_dds_update<-DESeq(mo1_dds_edit, test="LRT", reduced = ~1)
mo1_res_update<-results(mo1_dds_update)

sig_res_LRT_mo1<-mo1_res_update %>%
                  data.frame() %>%
                  rownames_to_column(var="gene")%>%
                  as_tibble() %>%
                  filter(padj < 0.015)

clustering_sig_res_LRT_mo1<-sig_res_LRT_mo1 %>%
  arrange(padj)

###Write P-values of LRT to file###
write.csv(as.data.frame(cbind(rownames(mo1_res_update),mo1_res_update$padj)), "mo1_LRT_padj_values_0.015.csv")


###Log2 Transform Count Data###
rld_mo1_dds<-rlog(mo1_dds_edit, blind=FALSE)
rld_mat_mo1_dds<-assay(rld_mo1_dds)

###Hierarchically Cluster Significant Differentially Expressed Genes###
cluster_rlog_sig_res_LRT_mo1<-rld_mat_mo1_dds[clustering_sig_res_LRT_mo1$gene, ]
clusters_sig_res_LRT_mo1<-degPatterns(cluster_rlog_sig_res_LRT_mo1, metadata = mo1_meta, col = "timepoint", plot = T, minc = 1)

###Calculate Z-score of Mo1 DE Genes###
cluster_rlog_sig_res_LRT_mo1_zscore<-t(scale(t(cluster_rlog_sig_res_LRT_mo1), scale=TRUE, center=TRUE))

LRT_mo1_sampling_timepoints<-c(-5,-5,-5,15,15,15,60,60,60,360,360,360,735,735,735,780,780,1080,1080,1080)

clusters_sig_res_LRT_mo1_vector<-clusters_sig_res_LRT_mo1$df$cluster
names(clusters_sig_res_LRT_mo1_vector)<-clusters_sig_res_LRT_mo1$df$genes

###Write Mo1 Cluster Numbers to file###
write.csv(as.data.frame(cbind(rownames(clusters_sig_res_LRT_mo1_vector),clusters_sig_res_LRT_mo1_vector)), "mo1_LRT_cluster_number_pvalue_0.015.csv")

###Format Data to Plot in ggplot###
LRT_mo1_data <- data.table(melt(data.table(class = as.factor(clusters_sig_res_LRT_mo1_vector), cluster_rlog_sig_res_LRT_mo1_zscore)))
LRT_mo1_data[, Time := rep(LRT_mo1_sampling_timepoints, each = nrow(cluster_rlog_sig_res_LRT_mo1_zscore)) ]
LRT_mo1_data[, ID := rep(1:nrow(cluster_rlog_sig_res_LRT_mo1_zscore), ncol(cluster_rlog_sig_res_LRT_mo1_zscore))]

LRT_mo1_data$facet = factor(LRT_mo1_data$class, levels = c("1", "6", "14","10"," ","2", "7", "9","11", "5","4", "12","13","","_","3","8"))
LRT_mo1_data$Time<-as.factor(LRT_mo1_data$Time)

panel.LRT_labs_mo1<- c("A.", "B.","C.", "D.", "E.", "F.", "G.", "H.", "I.", "J.", "K.", "L.", "M."," "," "," "," ","N.")
names(panel.LRT_labs_mo1)<-c("1", "6", "14","10","4", "12","2", "7", "9","11", "3", "8","5","","_","__","___","13")

LRT_panel_labeller_mo1 <- function(variable,value){
  return(panel.LRT_labs_mo1[value])
}

###Summarize Clusters and Format Summary for Plotting###
lrt_summary_mo1<-summarySE(LRT_mo1_data, measurevar="value", groupvars=c("class","Time"))
lrt_summary_mo1$Time=as.numeric(levels(lrt_summary_mo1$Time))[lrt_summary_mo1$Time]
lrt_summary_mo1$facet = factor(lrt_summary_mo1$class, levels = c("1", "6", "14","10"," ","2", "7", "9","11", "5","4", "12","13","","_","3","8"))
lrt_summary_mo1$annotation = c("N = 177",rep("",6), "N = 168",rep("",6),"N = 73",rep("",6), "N = 59",rep("",6), "N = 14",rep("",6), "N = 21",rep("",6), "N = 28",rep("",6), "N = 47",rep("",6), "N = 23",rep("",6), "N = 23",rep("",6), "N = 29",rep("",6), "N = 5",rep("",6), "N = 21",rep("",6),"N = 10",rep("",6))
lrt_summary_mo1$gene_group = c("A",rep("",6), "E",rep("",6),"M",rep("",6), "J",rep("",6), "I",rep("",6), "B",rep("",6), "F",rep("",6), "N",rep("",6), "G",rep("",6), "D",rep("",6), "H",rep("",6), "K",rep("",6), "L",rep("",6),"C",rep("",6))

###LINE PLOT OF HIERARCHICAL CLUSTERS###
lrt_summary_plot_mo1<-ggplot(lrt_summary_mo1, aes(x=Time, y=value)) +
  facet_wrap(~facet, ncol = 5, scales = "fixed", drop=FALSE) +
  geom_rect(data=NULL,aes(xmin=-Inf,xmax=0,ymin=-Inf,ymax=Inf), fill="gray")+
  geom_rect(data=NULL,aes(xmin=0,xmax=720,ymin=-Inf,ymax=Inf), fill="white")+
  geom_rect(data=NULL,aes(xmin=720,xmax=Inf,ymin=-Inf,ymax=Inf), fill="gray")+
  geom_ribbon(aes(ymin=value-sd, ymax=value+sd), alpha=.2) +
  geom_line() +
  geom_point()+
  labs(x = "Time (Hours)", y = "Z-score of transcript abundance") +
  scale_x_continuous(breaks=c(0,360,720,1080), labels = c("0","6","12","18"), position = "top") +
  theme(
  strip.text = element_blank(),
  text = element_text(size=25),
  legend.position = "none",
  strip.background =element_rect(fill="white"),
  panel.background = element_rect(fill="white"))+
  geom_text(aes(x=900, y=-2, label=annotation), size=6)+
  geom_text(aes(x=70, y=2, label=gene_group), size=8)
  

ggsave("mo1_LRT_clusters_line_plot_sd_ribbon_pvalue_0.015.tiff", width=85, height=55,device='tiff', units = "mm",dpi=300, scale=7)
```

Calculate Z-Score of rlog transformed counts

Generate Transcriptome Dynamics Plot for Ta8 and Mo1
```{r}
###TRANSCRIPTOME DYNAMICS###

#####PLOT Ta8 Z_SCORE DATA#####
ta8_zscore_table<-t(scale(t(rld_mat_ta8_dds), scale=TRUE, center=TRUE))
locustag_ta8<-substr(rownames(ta8_zscore_table),1,14)
ta8_all_locus_tags<-as.data.frame(locustag_ta8)
clusters_LRT_ta8_table<-clusters_sig_res_LRT_ta8_vector
clusters_LRT_ta8_table<-cbind(substr(names(clusters_sig_res_LRT_ta8_vector),1,14),clusters_LRT_ta8_table)
colnames(clusters_LRT_ta8_table)<-c("locustag_ta8","cluster_number")
ta8_all_locus_tags_with_clusters<-merge(ta8_all_locus_tags,clusters_LRT_ta8_table,all=TRUE)
ta8_all_locus_tags_with_clusters[is.na(ta8_all_locus_tags_with_clusters)] <- 0

ta8_all_locus_tags_with_clusters_vector<-ta8_all_locus_tags_with_clusters$cluster_number
names(ta8_all_locus_tags_with_clusters_vector)<-ta8_all_locus_tags_with_clusters$locustag_ta8

whole_LRT_ta8_data <- data.table(melt(data.table(class = as.factor(ta8_all_locus_tags_with_clusters_vector), ta8_zscore_table)))
whole_LRT_ta8_data[, Time := rep(LRT_ta8_sampling_timepoints, each = nrow(ta8_zscore_table)) ]
whole_LRT_ta8_data[, ID := rep(substr(rownames(ta8_zscore_table),1,14) , ncol(ta8_zscore_table))]

whole_LRT_ta8_data$facet = factor(whole_LRT_ta8_data$class, levels = c("0","9", "7", "8","5", "10","13", "1", "4","2", "3", "12", "6", "11", "14"))
whole_LRT_ta8_data$Time<-as.factor(whole_LRT_ta8_data$Time)

lrt_summary_ta8_for_wt_plot<-summarySE(whole_LRT_ta8_data, measurevar="value", groupvars=c("facet","Time","ID"))
lrt_summary_ta8_for_wt_plot$Time=as.numeric(levels(lrt_summary_ta8_for_wt_plot$Time))[lrt_summary_ta8_for_wt_plot$Time]
lrt_summary_ta8_for_wt_plot$facet = factor(lrt_summary_ta8_for_wt_plot$facet, levels = c("0","9", "7", "8","5", "10","13", "1", "4","2", "3", "12", "6", "11", "14"))

##TA8 PLOT###
ta8_transcriptome_rawcount<-ggplot(data=lrt_summary_ta8_for_wt_plot, aes(Time, value, group=ID))+
  geom_rect(data=NULL,aes(xmin=-Inf,xmax=0,ymin=-Inf,ymax=Inf), fill="light gray", color=NA)+
  geom_rect(data=NULL,aes(xmin=0,xmax=720,ymin=-Inf,ymax=Inf), fill="white", color=NA)+
  geom_rect(data=NULL,aes(xmin=720,xmax=Inf,ymin=-Inf,ymax=Inf), fill="light gray", color=NA)+
  theme_bw()+
  geom_line(data = subset(lrt_summary_ta8_for_wt_plot, facet == "0"),color="gray",size=1) +
  geom_line(data = subset(lrt_summary_ta8_for_wt_plot, facet == "8"),color="gray",size=1) +
  geom_line(data = subset(lrt_summary_ta8_for_wt_plot, facet == "5"),color="gray",size=1) +
  geom_line(data = subset(lrt_summary_ta8_for_wt_plot, facet == "2"),color="gray",size=1) +
  geom_line(data = subset(lrt_summary_ta8_for_wt_plot, facet == "3"),color="gray",size=1) +
  geom_line(data = subset(lrt_summary_ta8_for_wt_plot, facet == "9"),color="dark orange",size=1) +
  geom_line(data = subset(lrt_summary_ta8_for_wt_plot, facet == "7"),color="dark orange",size=1, alpha=0.6) +
  geom_line(data = subset(lrt_summary_ta8_for_wt_plot, facet == "8"),color="dark orange",size=1, alpha=0.6) +
  geom_line(data = subset(lrt_summary_ta8_for_wt_plot, facet == "5"),color="dark orange",size=1, alpha=0.6) +
  geom_line(data = subset(lrt_summary_ta8_for_wt_plot, facet == "1"),color="#3498DB",size=1, alpha=0.5) +
  geom_line(data = subset(lrt_summary_ta8_for_wt_plot, facet == "4"),color="#3498DB",size=1, alpha=0.5) +
  geom_line(data = subset(lrt_summary_ta8_for_wt_plot, facet == "2"),color="#3498DB",size=1, alpha=0.5) +
  geom_line(data = subset(lrt_summary_ta8_for_wt_plot, facet == "3"),color="#3498DB",size=1, alpha=0.5) +
  geom_line(data = subset(lrt_summary_ta8_for_wt_plot, facet == "11"),color="black",size=1, alpha=0.3) +
  geom_line(data = subset(lrt_summary_ta8_for_wt_plot, facet == "14"),color="black",size=1, alpha=0.3) +
  xlab("Time (Hours)")+
  ylab("Z-score of transcript abundance")+
  theme(text=element_text(size=30),axis.title.y = element_text(size=25),legend.position="none",legend.direction="vertical",legend.text=element_text(size=25))+
  scale_x_continuous(breaks=c(0,360,720,1080), labels = c("0","6","12","18")) +
  #scale_color_manual(values=c("red","#3498DB","#F7DC6F","black","red","red","#3498DB","#3498DB","#F7DC6F"))+
  ggtitle("A.")


#####PLOT Mo1 Z_SCORE DATA
mo1_zscore_table<-t(scale(t(rld_mat_mo1_dds), scale=TRUE, center=TRUE))
locustag_mo1<-substr(rownames(mo1_zscore_table),1,11)
mo1_all_locus_tags<-as.data.frame(locustag_mo1)
clusters_LRT_mo1_table<-clusters_sig_res_LRT_mo1_vector
clusters_LRT_mo1_table<-cbind(substr(names(clusters_sig_res_LRT_mo1_vector),1,11),clusters_LRT_mo1_table)
colnames(clusters_LRT_mo1_table)<-c("locustag_mo1","cluster_number")
mo1_all_locus_tags_with_clusters<-merge(mo1_all_locus_tags,clusters_LRT_mo1_table,all=TRUE)
mo1_all_locus_tags_with_clusters[is.na(mo1_all_locus_tags_with_clusters)] <- 0

mo1_all_locus_tags_with_clusters_vector<-mo1_all_locus_tags_with_clusters$cluster_number
names(mo1_all_locus_tags_with_clusters_vector)<-mo1_all_locus_tags_with_clusters$locustag_mo1

whole_LRT_mo1_data <- data.table(melt(data.table(class = as.factor(mo1_all_locus_tags_with_clusters_vector), mo1_zscore_table)))
whole_LRT_mo1_data[, Time := rep(LRT_mo1_sampling_timepoints, each = nrow(mo1_zscore_table)) ]
whole_LRT_mo1_data[, ID := rep(substr(rownames(mo1_zscore_table),1,11) , ncol(mo1_zscore_table))]

whole_LRT_mo1_data$facet = factor(whole_LRT_mo1_data$class, levels = c("0","1","6","14","10","2", "7", "9","11", "5","4", "12","13","3","8"))
whole_LRT_mo1_data$Time<-as.factor(whole_LRT_mo1_data$Time)

lrt_summary_mo1_for_wt_plot<-summarySE(whole_LRT_mo1_data, measurevar="value", groupvars=c("facet","Time","ID"))
lrt_summary_mo1_for_wt_plot$Time=as.numeric(levels(lrt_summary_mo1_for_wt_plot$Time))[lrt_summary_mo1_for_wt_plot$Time]
lrt_summary_mo1_for_wt_plot$facet = factor(lrt_summary_mo1_for_wt_plot$facet, levels = c("0","1","6","14","10","2", "7", "9","11", "5","4", "12","13","3","8"))

###Mo1 Z_SCORE Plot
mo1_transcriptome_rawcount<-ggplot(data=lrt_summary_mo1_for_wt_plot, aes(Time, value, group=ID))+
  geom_rect(data=NULL,aes(xmin=-Inf,xmax=0,ymin=-Inf,ymax=Inf), fill="light gray", color=NA)+
  geom_rect(data=NULL,aes(xmin=0,xmax=720,ymin=-Inf,ymax=Inf), fill="white", color=NA)+
  geom_rect(data=NULL,aes(xmin=720,xmax=Inf,ymin=-Inf,ymax=Inf), fill="light gray", color=NA)+
  theme_bw()+
  geom_line(data = subset(lrt_summary_mo1_for_wt_plot, facet == "0"),color="gray",size=1) +
  geom_line(data = subset(lrt_summary_mo1_for_wt_plot, facet == "1"),color="dark orange",size=1) +
  geom_line(data = subset(lrt_summary_mo1_for_wt_plot, facet == "6"),color="dark orange",size=1, alpha=0.5) +
  geom_line(data = subset(lrt_summary_mo1_for_wt_plot, facet == "14"),color="dark orange",size=1, alpha=0.5) +
  geom_line(data = subset(lrt_summary_mo1_for_wt_plot, facet == "10"),color="dark orange",size=1, alpha=0.5) +
  geom_line(data = subset(lrt_summary_mo1_for_wt_plot, facet == "2"),color="#3498DB",size=1, alpha=0.4) +
  geom_line(data = subset(lrt_summary_mo1_for_wt_plot, facet == "7"),color="#3498DB",size=1, alpha=0.4) +
  geom_line(data = subset(lrt_summary_mo1_for_wt_plot, facet == "9"),color="#3498DB",size=1, alpha=0.4) +
  geom_line(data = subset(lrt_summary_mo1_for_wt_plot, facet == "11"),color="#3498DB",size=1, alpha=0.4) +
  geom_line(data = subset(lrt_summary_mo1_for_wt_plot, facet == "5"),color="#3498DB",size=1, alpha=0.4) +
  geom_line(data = subset(lrt_summary_mo1_for_wt_plot, facet == "4"),color="yellow",size=1, alpha=0.3) +
  geom_line(data = subset(lrt_summary_mo1_for_wt_plot, facet == "12"),color="yellow",size=1, alpha=0.3) +
  geom_line(data = subset(lrt_summary_mo1_for_wt_plot, facet == "13"),color="yellow",size=1, alpha=0.3) +
  geom_line(data = subset(lrt_summary_mo1_for_wt_plot, facet == "3"),color="black",size=1, alpha=0.2) +
  geom_line(data = subset(lrt_summary_mo1_for_wt_plot, facet == "8"),color="black",size=1, alpha=0.2) +
  xlab("Time (Hours)")+
  ylab("Z-score of transcript abundance")+
  #ylab("Log2(Normalized Counts + 1) - Row Mean")+
  theme(text=element_text(size=30),axis.title.y = element_text(size=25),legend.position="none",legend.direction="vertical",legend.text=element_text(size=25))+
  scale_x_continuous(breaks=c(0,360,720,1080), labels = c("0","6","12","18")) +
  # scale_color_manual(values=c("grey","dark orange","#3498DB","black","yellow","#3498DB","dark orange","#3498DB","black","#3498DB","dark orange", "#3498DB","yellow","yellow", "dark orange"))+
  # scale_color_manual(values=c('0'="grey",'1'="dark orange",'6'="dark orange",'14'="dark orange",'10'="dark orange",'2'="#3498DB",'7'="#3498DB",'9'="#3498DB",'11'="#3498DB",'5'="#3498DB",'4'="yellow",'12'="yellow", '13'="yellow",'3'="black",'8'="black"))+
  ggtitle("B.")

ta8_transcriptome_rawcount/mo1_transcriptome_rawcount
ggsave("Mo1_Ta8_zscore_whole_patterns.tiff", width=85, height = 125,device='tiff', units = "mm",dpi=300, scale=3)

```

Generate Ta8 Known Light Sensing Gene Expression Plot
```{r}
df_ta8_lightsensors<-setDT(as.data.frame(df_ta8), keep.rownames = "Annotation")
melted_ta8ddsColl <- data.table(melt(data.table(class = as.factor(df_ta8_lightsensors$Annotation), df_ta8_lightsensors)))

names(melted_ta8ddsColl)[4]<-"foldchange"
melted_ta8ddsColl[, Time := rep(time_x__zero_var, each = nrow(df_ta8)) ]

lp_plot_counts<- subset(melted_ta8ddsColl, class=="Rhola_00012080:RHOLA_RS06005:Bacteriorhodopsin")
lp_plot_counts<-rbind(lp_plot_counts,subset(melted_ta8ddsColl, class=="Rhola_00011000:RHOLA_RS05465:deoxyribodipyrimidine photo-lyase"))
lp_plot_counts<-rbind(lp_plot_counts,subset(melted_ta8ddsColl, class=="Rhola_00013030:RHOLA_RS06470:deoxyribodipyrimidine photolyase-related protein"))
lp_plot_counts<-rbind(lp_plot_counts,subset(melted_ta8ddsColl, class=="Rhola_00007890:RHOLA_RS03920:deoxyribodipyrimidine photo-lyase"))

lp_plot_counts_ta8_zscore<- subset(lrt_summary_ta8_for_wt_plot, ID=="Rhola_00012080")
lp_plot_counts_ta8_zscore<-rbind(lp_plot_counts_ta8_zscore,subset(lrt_summary_ta8_for_wt_plot, ID=="Rhola_00011000"))
lp_plot_counts_ta8_zscore<-rbind(lp_plot_counts_ta8_zscore,subset(lrt_summary_ta8_for_wt_plot, ID=="Rhola_00013030"))
lp_plot_counts_ta8_zscore<-rbind(lp_plot_counts_ta8_zscore,subset(lrt_summary_ta8_for_wt_plot, ID=="Rhola_00007890"))


pd <- position_dodge(0.5) # move them .05 to the left and right

lightsensorsta8<-ggplot(data=lp_plot_counts_ta8_zscore, aes(x=Time, y=value, colour=ID, group=ID, fill=ID))+
  geom_rect(data=NULL,aes(xmin=-Inf,xmax=0,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  geom_rect(data=NULL,aes(xmin=0,xmax=720,ymin=-Inf,ymax=Inf), fill="white", color=NA)+
  geom_rect(data=NULL,aes(xmin=720,xmax=Inf,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  theme_bw()+
  xlab("Time (Hours)")+
  ylab("Z-score of transcript abundance")+
  theme(text=element_text(size=30),axis.title.y = element_text(size=25),legend.position="right",legend.direction="vertical",legend.text=element_text(size=25))+
  scale_x_continuous(breaks=c(0,360,720,1080), labels = c("0","6","12","18")) +
  scale_color_manual(values=c("purple","black","red", "blue"), labels = c("phrB", "cpf2", "actR", "cryB"), name="Gene")+
  scale_fill_manual(values=c("purple","black","white", "blue"), labels = c("phrB", "cpf2", "actR", "cryB"), name="Gene")+
  #geom_errorbar(aes(ymin=foldchange-SE, ymax=foldchange+SE), width=.2, position=position_jitter(width=2), size=0.5)+
  geom_line(size=1) +
  #geom_ribbon(aes(ymin=value-se, ymax=value+se), alpha=.2, color=NA)+
  geom_point(size=4,aes(shape=ID),stroke=1.5)+
  scale_shape_manual(values=c(24,23,21,22), labels = c("phrB", "cpf2", "actR", "cryB"), name="Gene")+
  ggtitle("A.")

```

Generage Mo1 Known Light Sensing Gene Expression Plots, 

Plot Ta8 and Mo1 Light Sensors Side by Side
```{r}
mo1ddsColl<-collapseReplicates(mo1_dds_edit, mo1_dds_edit@colData$condition)
norm_counts_mo1<-counts(mo1ddsColl, normalized=TRUE)
norm_counts_mo1<-log2(norm_counts_mo1)
norm_counts_mo1<-norm_counts_mo1[,c(1,3,5,7,2,4,6)]
norm_counts_mo1<-as.data.frame(norm_counts_mo1)

df_mo1_lightsensors<-setDT(as.data.frame(df_mo1), keep.rownames = "Annotation")

melted_mo1ddsColl <- data.table(melt(data.table(class = as.factor(df_mo1_lightsensors$Annotation), df_mo1_lightsensors)))


names(melted_mo1ddsColl)[4]<-"foldchange"
melted_mo1ddsColl[, Time := rep(time_x__zero_var, each = nrow(norm_counts_mo1)) ]

lp_plot_counts_mo1<- subset(melted_mo1ddsColl, class=="AURMO_00962:deoxyribodipyrimidine photolyase-related protein")
lp_plot_counts_mo1<-rbind(lp_plot_counts_mo1,subset(melted_mo1ddsColl, class=="AURMO_01673:deoxyribodipyrimidine photo-lyase type I"))
lp_plot_counts_mo1<-rbind(lp_plot_counts_mo1,subset(melted_mo1ddsColl, class=="AURMO_01564:hypothetical protein"))

lp_plot_counts_mo1_zscore<- subset(lrt_summary_mo1_for_wt_plot, ID=="AURMO_00962")
lp_plot_counts_mo1_zscore<-rbind(lp_plot_counts_mo1_zscore,subset(lrt_summary_mo1_for_wt_plot, ID=="AURMO_01673"))
lp_plot_counts_mo1_zscore<-rbind(lp_plot_counts_mo1_zscore,subset(lrt_summary_mo1_for_wt_plot, ID=="AURMO_01564"))


pd <- position_dodge(0.5) # move them .05 to the left and right

lightsensorsmo1<-ggplot(data=lp_plot_counts_mo1_zscore, aes(x=Time, y=value, colour=ID, group=ID, fill=ID))+
  geom_rect(data=NULL,aes(xmin=-Inf,xmax=0,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  geom_rect(data=NULL,aes(xmin=0,xmax=720,ymin=-Inf,ymax=Inf), fill="white", color=NA)+
  geom_rect(data=NULL,aes(xmin=720,xmax=Inf,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  theme_bw()+
  xlab("Time (Hours)")+
  ylab("Z-score of transcript abundance")+
  theme(text=element_text(size=30),axis.title.y = element_text(size=25),legend.position="right",legend.direction="vertical",legend.text=element_text(size=25))+
  scale_x_continuous(breaks=c(0,360,720,1080), labels = c("0","6","12","18")) +
  scale_color_manual(values=c("blue","dark green","purple"), labels=c('cryB',"heR","phrB"),name="Gene")+
  #geom_errorbar(aes(ymin=foldchange-SE, ymax=foldchange+SE), width=.2, position=position_jitter(width=2), size=0.5)+
  #geom_ribbon(aes(ymin=value-se, ymax=value+se), alpha=.2, color=NA)+
  scale_fill_manual(values=c("blue","dark green","purple"), labels=c('cryB',"heR","phrB"),name="Gene")+
  geom_line(size=1) +
  geom_point(size=4, aes(shape=ID),stroke=1.5)+
  scale_shape_manual(values=c(22,21,24), labels=c('cryB',"heR","phrB"),name="Gene")+
  ggtitle("B.")

(lightsensorsta8|lightsensorsmo1)

ggsave("lighsensors_z_score.tiff", width=85, height = 34, device='tiff', units = "mm",dpi=300, scale=7)

```

Competence Gene Expression Plots
```{r}
competence_ta8_zscore<- subset(lrt_summary_ta8_for_wt_plot, ID %in% c("Rhola_00009370","Rhola_00009380","Rhola_00009390","Rhola_00009400","Rhola_00003210","Rhola_00003220","Rhola_00003230","Rhola_00006080","Rhola_00006090","Rhola_00001130","Rhola_00005180"))

competence_mo1_zscore<- subset(lrt_summary_mo1_for_wt_plot, ID %in% c("AURMO_00482","AURMO_00483","AURMO_00484","AURMO_00485","AURMO_00486","AURMO_00316","AURMO_00317","AURMO_00318","AURMO_00910","AURMO_00911","AURMO_00599"))

ta8_competence_genes_plot<-ggplot(data=competence_ta8_zscore, aes(x=Time, y=value, colour=ID, group=ID, fill=ID))+
  geom_rect(data=NULL,aes(xmin=-Inf,xmax=0,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  geom_rect(data=NULL,aes(xmin=0,xmax=720,ymin=-Inf,ymax=Inf), fill="white", color=NA)+
  geom_rect(data=NULL,aes(xmin=720,xmax=Inf,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  theme_bw()+
  xlab("Time (Hours)")+
  ylab("Z-score of transcript abundance")+
  theme(text=element_text(size=30),axis.title.y = element_text(size=25),legend.position="right",legend.direction="vertical",legend.text=element_text(size=25))+
  scale_x_continuous(breaks=c(0,360,720,1080), labels = c("0","6","12","18")) +
  #scale_color_manual(values=c(rep("purple",1),rep("#CC79A7",3),rep("green",1),rep("#56B4E9",2),rep("black",4)), labels=c("comF","tadA2","tadBC","flp","dprA","comEA","comEC","HP","tadC","tadB/gspF","tadA1/cpaF"),name="Gene")+
  #geom_ribbon(aes(ymin=value-se, ymax=value+se), alpha=.2, color=NA)+
  #scale_fill_manual(values=c(rep("white",1),rep("#CC79A7",3),rep("white",1),rep("#56B4E9",2),rep("black",2),rep("white",2)), labels=c("comF","tadA2","tadBC","flp","dprA","comEA","comEC","HP","tadC","tadB/gspF","tadA1/cpaF"),name="Gene")+
  geom_line(size=1) +
  geom_point(size=5, aes(shape=ID), stroke=1.5)+
  #scale_shape_manual(values=c(22,22,12,7,21,21,13,24,25,24,25), labels=c("comF","tadA2","tadBC","flp","dprA","comEA","comEC","HP","tadC","tadB/gspF","tadA1/cpaF"),name="Gene")+
  ggtitle("A.")


mo1_competence_genes_plot<-ggplot(data=competence_mo1_zscore, aes(x=Time, y=value, colour=ID, group=ID, fill=ID))+
  geom_rect(data=NULL,aes(xmin=-Inf,xmax=0,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  geom_rect(data=NULL,aes(xmin=0,xmax=720,ymin=-Inf,ymax=Inf), fill="white", color=NA)+
  geom_rect(data=NULL,aes(xmin=720,xmax=Inf,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  theme_bw()+
  xlab("Time (Hours)")+
  ylab("Z-score of transcript abundance")+
  theme(text=element_text(size=30),axis.title.y = element_text(size=25),legend.position="right",legend.direction="vertical",legend.text=element_text(size=25))+
  scale_x_continuous(breaks=c(0,360,720,1080), labels = c("0","6","12","18")) +
  scale_color_manual(values=c(rep("#D55E00",3),rep("black",5),rep("green",1),rep("#56B4E9",2)), labels=c("tadA2","tadBC","flp","tadA1/cpaF","tadB","tadC","HP","tadE","dprA","comEC","comEA"),name="Gene")+
  #@geom_ribbon(aes(ymin=value-se, ymax=value+se), alpha=.2, color=NA)+
  scale_fill_manual(values=c(rep("#D55E00",3),rep("black",1),rep("white",4),rep("green",1),rep("#56B4E9",2)), labels=c("tadA2","tadBC","flp","tadA1/cpaF","tadB","tadC","HP","tadE","dprA","comEC","comEA"),name="Gene")+
  geom_line(size=1) +
  geom_point(size=5, aes(shape=ID),stroke=1.5)+
  scale_shape_manual(values=c(16,1,13,15,22,12,7,14,23,24,25), labels=c("tadA2","tadBC","flp","tadA1/cpaF","tadB","tadC","HP","tadE","dprA","comEC","comEA"),name="Gene")+
  ggtitle("B.")

ta8_competence_genes_plot+mo1_competence_genes_plot
ggsave("competence_genes_z_score.tiff", width=85, height = 34, device='tiff', units = "mm",dpi=300, scale=7)

```

Riboflavin Biosynthesis Gene Expression Plots
```{r}
riboflavin_ta8_zscore<- subset(lrt_summary_ta8_for_wt_plot, ID %in% c("Rhola_00011600","Rhola_00011610","Rhola_00011620","Rhola_00011630"))

riboflavin_mo1_zscore<- subset(lrt_summary_mo1_for_wt_plot, ID %in% c("AURMO_00338","AURMO_00339","AURMO_00340","AURMO_00341"))

ta8_riboflavin_genes_plot<-ggplot(data=riboflavin_ta8_zscore, aes(x=Time, y=value, colour=ID, group=ID, fill=ID))+
  geom_rect(data=NULL,aes(xmin=-Inf,xmax=0,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  geom_rect(data=NULL,aes(xmin=0,xmax=720,ymin=-Inf,ymax=Inf), fill="white", color=NA)+
  geom_rect(data=NULL,aes(xmin=720,xmax=Inf,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  theme_bw()+
  xlab("Time (Hours)")+
  ylab("Z-score of transcript abundance")+
  theme(text=element_text(size=30),axis.title.y = element_text(size=25),legend.position="right",legend.direction="vertical",legend.text=element_text(size=25))+
  scale_x_continuous(breaks=c(0,360,720,1080), labels = c("0","6","12","18")) +
  scale_color_manual(values=c(rep("#CC79A7",4)), labels = c("ribD","ribE","ribAB","ribH"),name="Gene")+
  #geom_ribbon(aes(ymin=value-se, ymax=value+se), alpha=.2, color=NA)+
  scale_fill_manual(values=c(rep("#CC79A7",4)), labels = c("ribD","ribE","ribAB","ribH"),name="Gene")+
  geom_line(size=1) +
  geom_point(size=5, aes(shape=ID))+
  scale_shape_manual(values=c(15,16,17,18), labels = c("ribD","ribE","ribAB","ribH"), name="Gene")+
  ggtitle("A.")



mo1_riboflavin_genes_plot<-ggplot(data=riboflavin_mo1_zscore, aes(x=Time, y=value, colour=ID, group=ID, fill=ID))+
  geom_rect(data=NULL,aes(xmin=-Inf,xmax=0,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  geom_rect(data=NULL,aes(xmin=0,xmax=720,ymin=-Inf,ymax=Inf), fill="white", color=NA)+
  geom_rect(data=NULL,aes(xmin=720,xmax=Inf,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  theme_bw()+
  xlab("Time (Hours)")+
  ylab("Z-score of transcript abundance")+
  theme(text=element_text(size=30),axis.title.y = element_text(size=25),legend.position="right",legend.direction="vertical",legend.text=element_text(size=25))+
  scale_x_continuous(breaks=c(0,360,720,1080), labels = c("0","6","12","18")) +
  scale_color_manual(values=c(rep("#D55E00",4)), labels = c("ribH","ribAB","ribE","ribD"),name="Gene")+
  #geom_ribbon(aes(ymin=value-sd, ymax=value+sd), alpha=.2, color=NA)+
  scale_fill_manual(values=c(rep("#D55E00",4)), labels = c("ribH","ribAB","ribE","ribD"),name="Gene")+
  geom_line(size=1) +
  geom_point(size=5, aes(shape=ID))+
  scale_shape_manual(values=c(15,16,17,18), labels = c("ribH","ribAB","ribE","ribD"),name="Gene")+
  ggtitle("B.")


ta8_riboflavin_genes_plot+mo1_riboflavin_genes_plot


ggsave("riboflavin_genes_z_score.tiff", width=85, height = 34, device='tiff', units = "mm",dpi=300, scale=7)

```

Starch Synthase Gene Expression Plots
```{r}
starch_synthase_ta8_zscore<- subset(lrt_summary_ta8_for_wt_plot, ID %in% c("Rhola_00005360","Rhola_00008040"))

starch_synthase_mo1_zscore<- subset(lrt_summary_mo1_for_wt_plot, ID %in% c("AURMO_01094","AURMO_00890"))

ta8_starch_synthase_genes_plot<-ggplot(data=starch_synthase_ta8_zscore, aes(x=Time, y=value, colour=ID, group=ID, fill=ID))+
  geom_rect(data=NULL,aes(xmin=-Inf,xmax=0,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  geom_rect(data=NULL,aes(xmin=0,xmax=720,ymin=-Inf,ymax=Inf), fill="white", color=NA)+
  geom_rect(data=NULL,aes(xmin=720,xmax=Inf,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  theme_bw()+
  xlab("Time (Hours)")+
  ylab("Z-score of transcript abundance")+
  theme(text=element_text(size=30),axis.title.y = element_text(size=25),legend.position="right",legend.direction="vertical",legend.text=element_text(size=25))+
  scale_x_continuous(breaks=c(0,360,720,1080), labels = c("0","6","12","18")) +
  scale_color_manual(values=c(rep("#CC79A7",2)), name="Gene")+
  #geom_ribbon(aes(ymin=value-se, ymax=value+se), alpha=.2, color=NA)+
  scale_fill_manual(values=c(rep("#CC79A7",2)), name="Gene")+
  geom_line(size=1) +
  geom_point(size=5, aes(shape=ID))+
  scale_shape_manual(values=c(15,16), name="Gene")+
  ggtitle("A.")



mo1_starch_synthase_genes_plot<-ggplot(data=starch_synthase_mo1_zscore, aes(x=Time, y=value, colour=ID, group=ID, fill=ID))+
  geom_rect(data=NULL,aes(xmin=-Inf,xmax=0,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  geom_rect(data=NULL,aes(xmin=0,xmax=720,ymin=-Inf,ymax=Inf), fill="white", color=NA)+
  geom_rect(data=NULL,aes(xmin=720,xmax=Inf,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  theme_bw()+
  xlab("Time (Hours)")+
  ylab("Z-score of transcript abundance")+
  theme(text=element_text(size=30),axis.title.y = element_text(size=25),legend.position="right",legend.direction="vertical",legend.text=element_text(size=25))+
  scale_x_continuous(breaks=c(0,360,720,1080), labels = c("0","6","12","18")) +
  scale_color_manual(values=c(rep("#D55E00",2)), name="Gene")+
  #@geom_ribbon(aes(ymin=value-se, ymax=value+se), alpha=.2, color=NA)+
  scale_fill_manual(values=c(rep("#D55E00",2)), name="Gene")+
  geom_line(size=1) +
  geom_point(size=5, aes(shape=ID))+
  scale_shape_manual(values=c(15,16), name="Gene")+
  ggtitle("B.")

ta8_starch_synthase_genes_plot+mo1_starch_synthase_genes_plot
ggsave("starch_synthase_genes_z_score.tiff", width=85, height = 34, device='tiff', units = "mm",dpi=300, scale=7)
```

Carotenoid Biosynthesis Gene Expression Plots
```{r}
carotenoid_synthesis_ta8_zscore<- subset(lrt_summary_ta8_for_wt_plot, ID %in% c("Rhola_00010860","Rhola_00010870","Rhola_00010880","Rhola_00010890","Rhola_00010900","Rhola_00000870","Rhola_00000880"))

carotenoid_synthesis_mo1_zscore<- subset(lrt_summary_mo1_for_wt_plot, ID %in% c("AURMO_01714","AURMO_01713","AURMO_01712","AURMO_01711","AURMO_01710"))

ta8_carotenoid_synthesis_genes_plot<-ggplot(data=carotenoid_synthesis_ta8_zscore, aes(x=Time, y=value, colour=ID, group=ID, fill=ID))+
  geom_rect(data=NULL,aes(xmin=-Inf,xmax=0,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  geom_rect(data=NULL,aes(xmin=0,xmax=720,ymin=-Inf,ymax=Inf), fill="white", color=NA)+
  geom_rect(data=NULL,aes(xmin=720,xmax=Inf,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  theme_bw()+
  xlab("Time (Hours)")+
  ylab("Z-score of transcript abundance")+
  theme(text=element_text(size=30),axis.title.y = element_text(size=25),legend.position="right",legend.direction="vertical",legend.text=element_text(size=25))+
  scale_x_continuous(breaks=c(0,360,720,1080), labels = c("0","6","12","18")) +
  scale_shape_manual(values=c(22,21,22,21,24,25,23),labels = c("pytoene desaturase","prenyltransferase","crtB","crtI","crtL1","crtL2","hydroxylase"), name="Gene")+
  scale_color_manual(values=c(rep("#CC79A7",7)), labels = c("pytoene desaturase","prenyltransferase","crtB","crtI","crtL1","crtL2","hydroxylase"),name="Gene")+
  #geom_ribbon(aes(ymin=value-se, ymax=value+se), alpha=.2, color=NA)+
  scale_fill_manual(values=c(rep("white",2),rep("#CC79A7",5)),labels = c("pytoene desaturase","prenyltransferase","crtB","crtI","crtL1","crtL2","hydroxylase"), name="Gene")+
  geom_line(size=1) +
  geom_point(size=5, aes(shape=ID), stroke=1.5)+
  ggtitle("A.")



mo1_carotenoid_synthesis_genes_plot<-ggplot(data=carotenoid_synthesis_mo1_zscore, aes(x=Time, y=value, colour=ID, group=ID, fill=ID))+
  geom_rect(data=NULL,aes(xmin=-Inf,xmax=0,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  geom_rect(data=NULL,aes(xmin=0,xmax=720,ymin=-Inf,ymax=Inf), fill="white", color=NA)+
  geom_rect(data=NULL,aes(xmin=720,xmax=Inf,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  theme_bw()+
  xlab("Time (Hours)")+
  ylab("Z-score of transcript abundance")+
  theme(text=element_text(size=30),axis.title.y = element_text(size=25),legend.position="right",legend.direction="vertical",legend.text=element_text(size=25))+
  scale_x_continuous(breaks=c(0,360,720,1080), labels = c("0","6","12","18")) +
  scale_color_manual(values=c(rep("#D55E00",5)), labels = c("hydroxylase","crtL2","crtL1","crtI","crtB"),name="Gene")+
  #@geom_ribbon(aes(ymin=value-se, ymax=value+se), alpha=.2, color=NA)+
  scale_fill_manual(values=c(rep("#D55E00",5)), labels = c("hydroxylase","crtL2","crtL1","crtI","crtB"),name="Gene")+
  geom_line(size=1) +
  geom_point(size=5, aes(shape=ID))+
  scale_shape_manual(values=c(23,25,24,21,22), labels = c("hydroxylase","crtL2","crtL1","crtI","crtB"),name="Gene")+
  ggtitle("B.")

ta8_carotenoid_synthesis_genes_plot+mo1_carotenoid_synthesis_genes_plot
ggsave("carotenoid_synthesis_genes_z_score.tiff", width=85, height = 34, device='tiff', units = "mm",dpi=300, scale=7)
```

ATP Synthase Gene Expression Plots
```{r}
atp_synthesis_ta8_zscore<- subset(lrt_summary_ta8_for_wt_plot, ID %in% c("Rhola_00004200","Rhola_00004210","Rhola_00004220","Rhola_00004230","Rhola_00004240","Rhola_00004250","Rhola_00004260","Rhola_00004270"))

atp_synthesis_mo1_zscore<- subset(lrt_summary_mo1_for_wt_plot, ID %in% c("AURMO_00472","AURMO_00473","AURMO_00474","AURMO_00475","AURMO_00476","AURMO_00477","AURMO_00478","AURMO_00479"))

ta8_atp_synthesis_genes_plot<-ggplot(data=atp_synthesis_ta8_zscore, aes(x=Time, y=value, colour=ID, group=ID, fill=ID))+
  geom_rect(data=NULL,aes(xmin=-Inf,xmax=0,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  geom_rect(data=NULL,aes(xmin=0,xmax=720,ymin=-Inf,ymax=Inf), fill="white", color=NA)+
  geom_rect(data=NULL,aes(xmin=720,xmax=Inf,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  theme_bw()+
  xlab("Time (Hours)")+
  ylab("Z-score of transcript abundance")+
  theme(text=element_text(size=20),axis.title.y = element_text(size=20),legend.position="right",legend.direction="vertical",legend.text=element_text(size=20))+
  scale_x_continuous(breaks=c(0,360,720,1080), labels = c("0","6","12","18")) +
  scale_color_manual(values=c(rep("#CC79A7",8)), labels = c("subunit a","subunit c","subunit b","subunit delta","subunit alpha","subunit gamma","subunit beta","subunit epsilon"),name="ATP Synthase Gene")+
  #geom_ribbon(aes(ymin=value-se, ymax=value+se), alpha=.2, color=NA)+
  scale_fill_manual(values=c(rep("#CC79A7",4),rep("white",4)), labels = c("subunit a","subunit c","subunit b","subunit delta","subunit alpha","subunit gamma","subunit beta","subunit epsilon"),name="ATP Synthase Gene")+
  geom_line(size=1) +
  geom_point(size=5, aes(shape=ID),stroke=1.5)+
  scale_shape_manual(values=c(22,21,24,25,22,21,24,25), labels = c("subunit a","subunit c","subunit b","subunit delta","subunit alpha","subunit gamma","subunit beta","subunit epsilon"),name="ATP Synthase Gene")+
  ggtitle("A.")



mo1_atp_synthesis_genes_plot<-ggplot(data=atp_synthesis_mo1_zscore, aes(x=Time, y=value, colour=ID, group=ID, fill=ID))+
  geom_rect(data=NULL,aes(xmin=-Inf,xmax=0,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  geom_rect(data=NULL,aes(xmin=0,xmax=720,ymin=-Inf,ymax=Inf), fill="white", color=NA)+
  geom_rect(data=NULL,aes(xmin=720,xmax=Inf,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  theme_bw()+
  xlab("Time (Hours)")+
  ylab("Z-score of transcript abundance")+
  theme(text=element_text(size=20),axis.title.y = element_text(size=20),legend.position="right",legend.direction="vertical",legend.text=element_text(size=20))+
  scale_x_continuous(breaks=c(0,360,720,1080), labels = c("0","6","12","18")) +
  scale_color_manual(values=c(rep("#D55E00",8)), labels = c("subunit a","subunit c","subunit b","subunit delta","subunit alpha","subunit gamma","subunit beta","subunit epsilon"),name="ATP Synthase Gene")+
  #@geom_ribbon(aes(ymin=value-se, ymax=value+se), alpha=.2, color=NA)+
  scale_fill_manual(values=c(rep("#D55E00",4),rep("white",4)), labels = c("subunit a","subunit c","subunit b","subunit delta","subunit alpha","subunit gamma","subunit beta","subunit epsilon"),name="ATP Synthase Gene")+
  geom_line(size=1) +
  geom_point(size=5, aes(shape=ID), stroke=1.5)+
  scale_shape_manual(values=c(22,21,24,25,22,21,24,25), labels = c("subunit a","subunit c","subunit b","subunit delta","subunit alpha","subunit gamma","subunit beta","subunit epsilon"),name="ATP Synthase Gene")+
  ggtitle("B.")

ta8_atp_synthesis_genes_plot+mo1_atp_synthesis_genes_plot
ggsave("atp_synthesis_genes_z_score.tiff", width=85, height = 34, device='tiff', units = "mm",dpi=300, scale=7)
```

Carbonic Anhydrase Gene Expression Plots
```{r}
carbonic_anhydrase_ta8_zscore<- subset(lrt_summary_ta8_for_wt_plot, ID %in% c("Rhola_00009890"))

carbonic_anhydrase_mo1_zscore<- subset(lrt_summary_mo1_for_wt_plot, ID %in% c("AURMO_00971","AURMO_01261"))

ta8_carbonic_anhydrase_genes_plot<-ggplot(data=carbonic_anhydrase_ta8_zscore, aes(x=Time, y=value, colour=ID, group=ID, fill=ID))+
  geom_rect(data=NULL,aes(xmin=-Inf,xmax=0,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  geom_rect(data=NULL,aes(xmin=0,xmax=720,ymin=-Inf,ymax=Inf), fill="white", color=NA)+
  geom_rect(data=NULL,aes(xmin=720,xmax=Inf,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  theme_bw()+
  xlab("Time (Hours)")+
  ylab("Z-score of transcript abundance")+
  theme(text=element_text(size=30),axis.title.y = element_text(size=25),legend.position="right",legend.direction="vertical",legend.text=element_text(size=25))+
  scale_x_continuous(breaks=c(0,360,720,1080), labels = c("0","6","12","18")) +
  scale_color_manual(values=c(rep("#CC79A7",1)), name="Gene")+
  #geom_ribbon(aes(ymin=value-se, ymax=value+se), alpha=.2, color=NA)+
  scale_fill_manual(values=c(rep("#CC79A7",1)), name="Gene")+
  geom_line(size=1) +
  geom_point(size=5, aes(shape=ID))+
  scale_shape_manual(values=c(15), name="Gene")+
  ggtitle("A.")



mo1_carbonic_anhydrase_genes_plot<-ggplot(data=carbonic_anhydrase_mo1_zscore, aes(x=Time, y=value, colour=ID, group=ID, fill=ID))+
  geom_rect(data=NULL,aes(xmin=-Inf,xmax=0,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  geom_rect(data=NULL,aes(xmin=0,xmax=720,ymin=-Inf,ymax=Inf), fill="white", color=NA)+
  geom_rect(data=NULL,aes(xmin=720,xmax=Inf,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  theme_bw()+
  xlab("Time (Hours)")+
  ylab("Z-score of transcript abundance")+
  theme(text=element_text(size=30),axis.title.y = element_text(size=25),legend.position="right",legend.direction="vertical",legend.text=element_text(size=25))+
  scale_x_continuous(breaks=c(0,360,720,1080), labels = c("0","6","12","18")) +
  scale_color_manual(values=c(rep("#D55E00",2)), name="Gene")+
  #@geom_ribbon(aes(ymin=value-se, ymax=value+se), alpha=.2, color=NA)+
  scale_fill_manual(values=c(rep("#D55E00",2)), name="Gene")+
  geom_line(size=1) +
  geom_point(size=5, aes(shape=ID))+
  scale_shape_manual(values=c(15,16), name="Gene")+
  ggtitle("B.")

ta8_carbonic_anhydrase_genes_plot+mo1_carbonic_anhydrase_genes_plot
ggsave("carbonic_anhydrase_genes_z_score.tiff", width=85, height = 34, device='tiff', units = "mm",dpi=300, scale=7)
```

Peroxiredoxin Gene Expression Plots
```{r}
peroxiredoxin_ta8_zscore<- subset(lrt_summary_ta8_for_wt_plot, ID %in% c("Rhola_00006540","Rhola_00011030"))

peroxiredoxin_mo1_zscore<- subset(lrt_summary_mo1_for_wt_plot, ID %in% c("AURMO_00420","AURMO_00726"))

ta8_peroxiredoxin_genes_plot<-ggplot(data=peroxiredoxin_ta8_zscore, aes(x=Time, y=value, colour=ID, group=ID, fill=ID))+
  geom_rect(data=NULL,aes(xmin=-Inf,xmax=0,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  geom_rect(data=NULL,aes(xmin=0,xmax=720,ymin=-Inf,ymax=Inf), fill="white", color=NA)+
  geom_rect(data=NULL,aes(xmin=720,xmax=Inf,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  theme_bw()+
  xlab("Time (Hours)")+
  ylab("Z-score of transcript abundance")+
  theme(text=element_text(size=30),axis.title.y = element_text(size=25),legend.position="right",legend.direction="vertical",legend.text=element_text(size=25))+
  scale_x_continuous(breaks=c(0,360,720,1080), labels = c("0","6","12","18")) +
  scale_color_manual(values=c(rep("#CC79A7",2)), name="Gene")+
  #geom_ribbon(aes(ymin=value-se, ymax=value+se), alpha=.2, color=NA)+
  scale_fill_manual(values=c(rep("#CC79A7",2)), name="Gene")+
  geom_line(size=1) +
  geom_point(size=5, aes(shape=ID))+
  scale_shape_manual(values=c(15,16), name="Gene")+
  ggtitle("A.")



mo1_peroxiredoxin_genes_plot<-ggplot(data=peroxiredoxin_mo1_zscore, aes(x=Time, y=value, colour=ID, group=ID, fill=ID))+
  geom_rect(data=NULL,aes(xmin=-Inf,xmax=0,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  geom_rect(data=NULL,aes(xmin=0,xmax=720,ymin=-Inf,ymax=Inf), fill="white", color=NA)+
  geom_rect(data=NULL,aes(xmin=720,xmax=Inf,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  theme_bw()+
  xlab("Time (Hours)")+
  ylab("Z-score of transcript abundance")+
  theme(text=element_text(size=30),axis.title.y = element_text(size=25),legend.position="right",legend.direction="vertical",legend.text=element_text(size=25))+
  scale_x_continuous(breaks=c(0,360,720,1080), labels = c("0","6","12","18")) +
  scale_color_manual(values=c(rep("#D55E00",2)), name="Gene")+
  #@geom_ribbon(aes(ymin=value-se, ymax=value+se), alpha=.2, color=NA)+
  scale_fill_manual(values=c(rep("#D55E00",2)), name="Gene")+
  geom_line(size=1) +
  geom_point(size=5, aes(shape=ID))+
  scale_shape_manual(values=c(15,16), name="Gene")+
  ggtitle("B.")

ta8_peroxiredoxin_genes_plot+mo1_peroxiredoxin_genes_plot
ggsave("peroxiredoxin_genes_z_score.tiff", width=85, height = 34, device='tiff', units = "mm",dpi=300, scale=7)
```

TCA Cycle Gene Expression Plots
```{r}
tca_cycle_ta8_zscore<- subset(lrt_summary_ta8_for_wt_plot, ID %in% c("Rhola_00009130","Rhola_00006940","Rhola_00011790","Rhola_00006750","Rhola_00009530","Rhola_00011830","Rhola_00011840","Rhola_00011530","Rhola_00011540","Rhola_00011550","Rhola_00011560","Rhola_00007070"))

tca_cycle_mo1_zscore<- subset(lrt_summary_mo1_for_wt_plot, ID %in% c("AURMO_00993","AURMO_00762","AURMO_01577","AURMO_00429","AURMO_00746","AURMO_01515","AURMO_01516","AURMO_00354","AURMO_00355","AURMO_00356","AURMO_00357","AURMO_01558","AURMO_01262","AURMO_00302"))

ta8_tca_cycle_genes_plot<-ggplot(data=tca_cycle_ta8_zscore, aes(x=Time, y=value, colour=ID, group=ID, fill=ID))+
  geom_rect(data=NULL,aes(xmin=-Inf,xmax=0,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  geom_rect(data=NULL,aes(xmin=0,xmax=720,ymin=-Inf,ymax=Inf), fill="white", color=NA)+
  geom_rect(data=NULL,aes(xmin=720,xmax=Inf,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  theme_bw()+
  xlab("Time (Hours)")+
  ylab("Z-score of transcript abundance")+
  theme(text=element_text(size=30),axis.title.y = element_text(size=25),legend.position="right",legend.direction="vertical",legend.text=element_text(size=25))+
  scale_x_continuous(breaks=c(0,360,720,1080), labels = c("0","6","12","18")) +
  scale_color_manual(values=c(rep("#CC79A7",12)), name="Gene")+
  #geom_ribbon(aes(ymin=value-se, ymax=value+se), alpha=.2, color=NA)+
  scale_fill_manual(values=c(rep("#CC79A7",4),rep("white",5),rep("#CC79A7",3)), name="Gene")+
  geom_line(size=1) +
  geom_point(size=5, aes(shape=ID), stroke=1.5)+
  scale_shape_manual(values=c(22,21,24,23,22,21,24,23,25,10,12,7), name="Gene")+
  ggtitle("A.")



mo1_tca_cycle_genes_plot<-ggplot(data=tca_cycle_mo1_zscore, aes(x=Time, y=value, colour=ID, group=ID, fill=ID))+
  geom_rect(data=NULL,aes(xmin=-Inf,xmax=0,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  geom_rect(data=NULL,aes(xmin=0,xmax=720,ymin=-Inf,ymax=Inf), fill="white", color=NA)+
  geom_rect(data=NULL,aes(xmin=720,xmax=Inf,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  theme_bw()+
  xlab("Time (Hours)")+
  ylab("Z-score of transcript abundance")+
  theme(text=element_text(size=30),axis.title.y = element_text(size=25),legend.position="right",legend.direction="vertical",legend.text=element_text(size=25))+
  scale_x_continuous(breaks=c(0,360,720,1080), labels = c("0","6","12","18")) +
  scale_color_manual(values=c(rep("blue",1),rep("#D55E00",5),rep("black",1),rep("blue",1),rep("black",1),rep("#D55E00",1),rep("blue",4)), name="Gene")+
  #@geom_ribbon(aes(ymin=value-se, ymax=value+se), alpha=.2, color=NA)+
  scale_fill_manual(values=c(rep("blue",1),rep("#D55E00",3),rep("white",5),rep("#D55E00",1),rep("blue",4)), name="Gene")+
  geom_line(size=1) +
  geom_point(size=5, aes(shape=ID), stroke=1.5)+
  scale_shape_manual(values=c(22,21,24,23,22,21,24,23,25,10,13,12,7,14), name="Gene")+
  ggtitle("B.")

ta8_tca_cycle_genes_plot+mo1_tca_cycle_genes_plot
ggsave("tca_cycle_genes_z_score.tiff", width=85, height = 34, device='tiff', units = "mm",dpi=300, scale=7)
```

Rodocoxin Gene Expression Plots
```{r}

rhodocoxin_mo1_zscore<- subset(lrt_summary_mo1_for_wt_plot, ID %in% c("AURMO_01300","AURMO_01301","AURMO_01493","AURMO_01494","AURMO_01495"))

mo1_rhodocoxin_genes_plot<-ggplot(data=rhodocoxin_mo1_zscore, aes(x=Time, y=value, colour=ID, group=ID, fill=ID))+
  geom_rect(data=NULL,aes(xmin=-Inf,xmax=0,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  geom_rect(data=NULL,aes(xmin=0,xmax=720,ymin=-Inf,ymax=Inf), fill="white", color=NA)+
  geom_rect(data=NULL,aes(xmin=720,xmax=Inf,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  theme_bw()+
  xlab("Time (Hours)")+
  ylab("Z-score of transcript abundance")+
  theme(text=element_text(size=40),axis.title.y = element_text(size=35),legend.position="right",legend.direction="vertical",legend.text=element_text(size=35))+
  scale_x_continuous(breaks=c(0,360,720,1080), labels = c("0","6","12","18")) +
  scale_color_manual(values=c(rep("#D55E00",2),rep("black",3)), name="Gene")+
  #@geom_ribbon(aes(ymin=value-se, ymax=value+se), alpha=.2, color=NA)+
  scale_fill_manual(values=c(rep("#D55E00",2),rep("white",3)), name="Gene")+
  geom_line(size=1) +
  geom_point(size=7, aes(shape=ID), stroke=1.5)+
  scale_shape_manual(values=c(22,21,22,21,24), name="Gene")
  #ggtitle("B.")


ggsave("rhodocoxin_genes_z_score.tiff", width=85, height = 68, device='tiff', units = "mm",dpi=300, scale=7)
```

Restriction Enzyme Gene Expression Plots
```{r}
restriction_enzymes_ta8_zscore<- subset(lrt_summary_ta8_for_wt_plot, ID %in% c("Rhola_00001140","Rhola_00001160","Rhola_00001170","Rhola_00010350"))

restriction_enzymes_mo1_zscore<- subset(lrt_summary_mo1_for_wt_plot, ID %in% c("AURMO_00257","AURMO_00358","AURMO_01424"))

ta8_restriction_enzymes_genes_plot<-ggplot(data=restriction_enzymes_ta8_zscore, aes(x=Time, y=value, colour=ID, group=ID, fill=ID))+
  geom_rect(data=NULL,aes(xmin=-Inf,xmax=0,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  geom_rect(data=NULL,aes(xmin=0,xmax=720,ymin=-Inf,ymax=Inf), fill="white", color=NA)+
  geom_rect(data=NULL,aes(xmin=720,xmax=Inf,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  theme_bw()+
  xlab("Time (Hours)")+
  ylab("Z-score of transcript abundance")+
  theme(text=element_text(size=30),axis.title.y = element_text(size=25),legend.position="right",legend.direction="vertical",legend.text=element_text(size=25))+
  scale_x_continuous(breaks=c(0,360,720,1080), labels = c("0","6","12","18")) +
  scale_color_manual(values=c(rep("#CC79A7",4)), name="Gene")+
  #geom_ribbon(aes(ymin=value-se, ymax=value+se), alpha=.2, color=NA)+
  scale_fill_manual(values=c(rep("#CC79A7",4)), name="Gene")+
  geom_line(size=1) +
  geom_point(size=5, aes(shape=ID))+
  scale_shape_manual(values=c(15,16,17,18), name="Gene")+
  ggtitle("A.")



mo1_restriction_enzymes_genes_plot<-ggplot(data=restriction_enzymes_mo1_zscore, aes(x=Time, y=value, colour=ID, group=ID, fill=ID))+
  geom_rect(data=NULL,aes(xmin=-Inf,xmax=0,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  geom_rect(data=NULL,aes(xmin=0,xmax=720,ymin=-Inf,ymax=Inf), fill="white", color=NA)+
  geom_rect(data=NULL,aes(xmin=720,xmax=Inf,ymin=-Inf,ymax=Inf), fill="gray", color=NA)+
  theme_bw()+
  xlab("Time (Hours)")+
  ylab("Z-score of transcript abundance")+
  theme(text=element_text(size=30),axis.title.y = element_text(size=25),legend.position="right",legend.direction="vertical",legend.text=element_text(size=25))+
  scale_x_continuous(breaks=c(0,360,720,1080), labels = c("0","6","12","18")) +
  scale_color_manual(values=c(rep("#D55E00",3)), name="Gene")+
  #@geom_ribbon(aes(ymin=value-se, ymax=value+se), alpha=.2, color=NA)+
  scale_fill_manual(values=c(rep("#D55E00",3)), name="Gene")+
  geom_line(size=1) +
  geom_point(size=5, aes(shape=ID))+
  scale_shape_manual(values=c(15,16,17), name="Gene")+
  ggtitle("B.")

ta8_restriction_enzymes_genes_plot+mo1_restriction_enzymes_genes_plot
ggsave("restriction_enzymes_genes_z_score.tiff", width=85, height = 34, device='tiff', units = "mm",dpi=300, scale=7)
```

Print DESEQ2 Normalized Counts to an Excel File
```{r}
##GET NORMALIZED COUNTS##
write.csv(counts(ta8_dds, normalize=TRUE), file = "ta8_deseq_normalized_count.csv")
write.csv(counts(mo1_dds_edit, normalize=TRUE), file = "mo1_deseq_normalized_count.csv")
#########################
```

